src/msgpack.cpp

```diff
@@ -0,0 +1,155 @@
+#include <migraphx/msgpack.hpp>
+#include <migraphx/serialize.hpp>
+#include <msgpack.hpp>
+
+namespace msgpack {
+MSGPACK_API_VERSION_NAMESPACE(MSGPACK_DEFAULT_API_NS)
+{
+    namespace adaptor {
+
+    template <>
+    struct convert<migraphx::value>
+    {
+        const msgpack::object& operator()(const msgpack::object& o, migraphx::value& v) const
+        {
+            switch(o.type)
+            {
+            case msgpack::type::NIL:
+            {
+                v = nullptr;
+                break;
+            }
+            case msgpack::type::BOOLEAN:
+            {
+                v = o.as<bool>();
+                break;
+            }
+            case msgpack::type::POSITIVE_INTEGER:
+            {
+                v = o.as<std::uint64_t>();
+                break;
+            }
+            case msgpack::type::NEGATIVE_INTEGER:
+            {
+                v = o.as<std::int64_t>();
+                break;
+            }
+            case msgpack::type::FLOAT32:
+            case msgpack::type::FLOAT64:
+            {
+                v = o.as<double>();
+                break;
+            }
+            case msgpack::type::STR:
+            {
+                v = o.as<std::string>();
+                break;
+            }
+            case msgpack::type::BIN: { MIGRAPHX_THROW("msgpack BIN type not supported.");
+            }
+            case msgpack::type::ARRAY:
+            {
+                migraphx::value r = migraphx::value::array{};
+                std::for_each(
+                    o.via.array.ptr,
+                    o.via.array.ptr + o.via.array.size,
+                    [&](const msgpack::object& so) { r.push_back(so.as<migraphx::value>()); });
+                v = r;
+                break;
+            }
+            case msgpack::type::MAP:
+            {
+                migraphx::value r = migraphx::value::object{};
+                std::for_each(o.via.map.ptr,
+                              o.via.map.ptr + o.via.map.size,
+                              [&](const msgpack::object_kv& p) {
+                                  r[p.key.as<std::string>()] = p.val.as<migraphx::value>();
+                              });
+                v = r;
+                break;
+            }
+            case msgpack::type::EXT: { MIGRAPHX_THROW("msgpack BIN type not supported.");
```

Comment:
I think the msssage should be "msgpack EXT type not support", not BIN type

Reply:
Ah good catch!